GoldRush.js

```diff
@@ -0,0 +1,93 @@
+// const Matrix = require("./Matrix");
+
+class GoldRush extends Matrix {
+    constructor(rowNum, colNum) {
+        super(rowNum, colNum)
+        this.generateMatrix(rowNum, colNum)
+        this.addCoins(rowNum, colNum)
+        this.alter(0, 0, 1)
+        this.alter(4, 4, 2)
+        this.playerScores = {
+            1: 0,
+            2: 0
+        }
+    }
+
+
+    movePlayer(playerNum, direction) {
+        let x = this.findCoordinate(playerNum).y
+        let y = this.findCoordinate(playerNum).x
+        
+        if (direction == 'down'
```

Comment:
Too many conditions. Try abstracting the different conditions and creating helper functions.